python/cudf_polars/cudf_polars/utils/config.py

```diff
@@ -5,209 +5,306 @@
+    shuffle_method
+        The method to use for shuffling data between workers. Currently only
+        ``ShuffleMethod.TASKS`` is supported for the in-memory executor.
+    broadcast_join_limit
+        The maximum number of partitions to allow for the smaller table in
+        a broadcast join.
+    """
+
+    name: Literal["in-memory"] = dataclasses.field(default="in-memory", init=False)
+    scheduler: Literal["synchronous"] = "synchronous"
+    shuffle_method: Literal["tasks"] | None = None
+    broadcast_join_limit: int = 32
+
+    def __post_init__(self) -> None:
+        if self.scheduler != "synchronous":
             raise ValueError(
-                f"Unsupported executor_options for {executor}: {unsupported}"
+                "'synchronous' is the only valid scheduler for the in-memory executor"
             )
+        if self.shuffle_method is not None and self.shuffle_method != "tasks":
+            raise ValueError(
+                "'tasks' is the only valid shuffle method for the in-memory executor"
+            )
+
+        if not isinstance(self.broadcast_join_limit, int):
+            raise TypeError("broadcast_join_limit must be an int")
+
 
-        return config
-
-
-def _valid_option(
-    options: MutableMapping[str, Any],
-    key: str,
-    *,
-    default: Any,
-    choices: Sequence[Any] | None = None,
-    astype: type | None = None,
-) -> MutableMapping[str, Any]:
-    """Return a valid configuration option."""
-    value = options.get(key, default)
-    if astype is not None:
-        value = astype(value)
-    if choices is not None and value not in choices:
-        raise ValueError(f"Unsupported {key} option: {value}. Choices are: {choices}")
-    options[key] = value
-    return options
-
-
-def _valid_streaming_options(
-    options: MutableMapping[str, Any],
-) -> tuple[MutableMapping[str, Any], set[str]]:
-    """Return valid options for the streaming executor."""
-    default: Any
-    choices: Sequence[Any]
-    for key in (
-        _STREAMING_OPTIONS := (
-            "scheduler",  # Validate scheduler first
-            "fallback_mode",
-            "max_rows_per_partition",
-            "parquet_blocksize",
-            "cardinality_factor",
-            "groupby_n_ary",
-            "broadcast_join_limit",
-            "shuffle_method",
+@dataclasses.dataclass(frozen=True, eq=True)
+class ConfigOptions:
+    """
+    Configuration for the polars GPUEngine.
+
+    Parameters
+    ----------
+    raise_on_fail
+        Whether to raise an exception when the GPU engine fails to execute a
+        query. ``False`` by default.
+    parquet_options
+        Options controlling parquet file reading and writing. See
+        :class:`ParquetOptions` for more.
+    executor
+        The executor to use for the GPU engine. See :class:`StreamingExecutor`
+        and :class:`InMemoryExecutor` for more.
+    device
+        The device to use for the GPU engine.
```

Comment:
nitpick: From the polars docs
```suggestion
        The GPU used to run the query. If not provided, the
        query uses the current CUDA device.
```